var/spack/repos/builtin/packages/samtools/package.py

```diff
@@ -41,6 +41,7 @@ class Samtools(Package):
     depends_on("python", type="run")
 
     # htslib became standalone @1.3.1, must use corresponding version
+    depends_on("htslib@1.16", when="@1.16.1")
```

Comment:
@downloadico It looks like you added the dependency here, but not the new version :)